lib/galaxy/tools/__init__.py

```diff
@@ -479,7 +488,13 @@ def requires_galaxy_python_environment(self):
         elif preserve_python_environment == "legacy_and_local" and self.repository_id is None:
             return True
         else:
-            return self.old_id in GALAXY_LIB_TOOLS
+            unversioned_legacy_tool = self.old_id in GALAXY_LIB_TOOLS_UNVERSIONED
+            versioned_legacy_tool = self.old_id in GALAXY_LIB_TOOLS_VERSIONED
+            legacy_tool = unversioned_legacy_tool or versioned_legacy_tool
+            if not legacy_tool:
+                return False
+            else:
+                return unversioned_legacy_tool or self.version_object < GALAXY_LIB_TOOLS_VERSIONED[self.old_id]
```

Comment:
No problem - I'll take any excuse to populate the commit history with :eagle:s.

 :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: :eagle: